### PR TITLE
Ignore -native-compiler option when disabled

### LINKED ISSUE #39: Compilation broken if Coq compiled without native compiler
Apparently coqc is broken if Coq does not support native compilation but you pass the flag as per #38 . On windows we disable the native compiler (it is the default).

Relevant logs:
https://coq.gitlab.io/-/coq/-/jobs/501853273/artifacts/artifacts/buildlogs/bignums-a33927c06eb0701025f873f7e5951f7b07a89e6e-make_log.txt
https://coq.gitlab.io/-/coq/-/jobs/501853273/artifacts/artifacts/buildlogs/bignums-a33927c06eb0701025f873f7e5951f7b07a89e6e-make_err.txt

I report the bug here, but it is unclear to me, since clients have (AFAICT) no way to ask Coq if native compilation is supported.

CC @maximedenes

### PR DESCRIPTION
**Pull request by proux01**
Fixes coq/bignums#39

- [x] Added / updated test-suite
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

File: toplevel/coqargs.ml [hunk 1]
```diff
@@ -286,6 +286,29 @@ let parse_option_set opt =
     let v = String.sub opt (eqi+1) (len - eqi - 1) in
     to_opt_key (String.sub opt 0 eqi), Some v
 
+let warn_no_native_compiler =
+  let open Pp in
+  CWarnings.create ~name:"native-compiler-disabled" ~category:"native-compiler"
+         (fun () -> strbrk "Native compiler is disabled," ++
+                      strbrk " -native-compiler option ignored.")
```
Comment (SkySkimmer) [inline toplevel/coqargs.ml:293]:
```suggestion
  CWarnings.create ~name:"native-compiler-disabled" ~category:"native-compiler"
         Pp.(fun () -> strbrk "Native compiler is disabled," ++
                      strbrk " -native-compiler option ignored.")
```
this form is more common

File: toplevel/coqargs.ml [hunk 2]
```diff
@@ -286,6 +286,29 @@ let parse_option_set opt =
...
+
+let get_native_compiler s =
+  if not Coq_config.native_compiler then
+    let () = warn_no_native_compiler () in
+    default_native
```
Comment (SkySkimmer) [inline toplevel/coqargs.ml:298]: We shouldn't warn in the "no" case IMO

File: toplevel/coqargs.ml (hunk 1 above)
Comment (proux01) [reply toplevel/coqargs.ml:293]: Ok

File: toplevel/coqargs.ml (hunk 2 above)
Comment (proux01) [reply toplevel/coqargs.ml]: Ok

**Comment by ppedrot** [comment]
Why does the top post say that there is a changelog when the PR does not contain one? @proux01 did you forget to git-add it?

**Review by ppedrot — APPROVED** [review]
Modulo changelog LGTM

**Comment by proux01** [comment]
> Why does the top post say that there is a changelog when the PR does not contain one? @proux01 did you forget to git-add it?

Looks like, here it is. Sorry about that.

**Comment by ppedrot** [comment]
Will merge soon.

**Pull request merged**